### ISSUE TITLE
md-card - primary image css selector and border radius issues

### ISSUE THREAD

**Issue by guzza** [angular/material#1230]
The new changes to `md-card` introduce the following issues:

1. CSS for `<img>` element of `md-card` (primary action) selects by using a direct child selector and in that way hinders wrapping the `<img>` in a `<a>` element not giving us the possibility to make it clickable as per material design spec it should lead/do the primary action while the action below should be supplemental actions.

2. In the 0.6.1 version example of `md-card` a class `.md-card-image` was applied to the `<img>` in the example and that would implement a border-radius to the image but in the new ones that was dropped even tough the class still exists. That class is either redundant or needs to be updated since it only applies top two corners border radius which is a problem if the `md-card` only has the picture the bottom two corners push out the `md-card` border-radius to be a corner.

3. `md-card` border-radius can be pushed out by the inner elements (especially the `<img>`) and is very simillar to a previous issue I submitted regarding the `md-dialog` - #1015 and that should be resolved so `md-card` border-radius is always present.

**Comment by Frank3K** [comment]
+1 for class based rules (`.md-card-image`) instead of direct child base rules (`> img`).

**Comment by ajoslin** [comment]
The selectors are now:

```css
  > img,
  :not(md-card-content) img {
    order: 0;
    width: 100%;
  }
```

**Comment by guzza** [comment]
Hey you forgot to put in a `"overflow:hidden"` fix on `md-card` that is the same fix that was implemented for `md-dialog` in the issue #1015. Now the inner elements can push out the border-radius set on `md-card`.
